redwood-lazylayout-uiview/src/commonMain/kotlin/app/cash/redwood/lazylayout/uiview/UIViewLazyList.kt

```diff
@@ -25,133 +25,144 @@ import app.cash.redwood.Modifier
+import app.cash.redwood.lazylayout.widget.LazyList
+import app.cash.redwood.lazylayout.widget.LazyListUpdateProcessor
 import app.cash.redwood.lazylayout.widget.RefreshableLazyList
-import app.cash.redwood.lazylayout.widget.WindowedChildren
-import app.cash.redwood.lazylayout.widget.WindowedLazyList
 import app.cash.redwood.ui.Margin
 import app.cash.redwood.widget.ChangeListener
-import app.cash.redwood.widget.MutableListChildren
 import app.cash.redwood.widget.Widget
 import kotlinx.cinterop.CValue
 import kotlinx.cinterop.ObjCClass
+import kotlinx.cinterop.convert
 import kotlinx.cinterop.readValue
-import kotlinx.cinterop.useContents
-import platform.CoreGraphics.CGFloat
 import platform.CoreGraphics.CGRect
 import platform.CoreGraphics.CGRectZero
-import platform.CoreGraphics.CGSizeMake
+import platform.CoreGraphics.CGSize
 import platform.Foundation.NSIndexPath
 import platform.Foundation.classForCoder
-import platform.UIKit.UICollectionView
-import platform.UIKit.UICollectionViewCell
-import platform.UIKit.UICollectionViewDataSourceProtocol
-import platform.UIKit.UICollectionViewDelegateFlowLayoutProtocol
-import platform.UIKit.UICollectionViewFlowLayout
-import platform.UIKit.UICollectionViewFlowLayoutAutomaticSize
-import platform.UIKit.UICollectionViewLayout
-import platform.UIKit.UICollectionViewLayoutAttributes
-import platform.UIKit.UICollectionViewScrollDirection.UICollectionViewScrollDirectionHorizontal
-import platform.UIKit.UICollectionViewScrollDirection.UICollectionViewScrollDirectionVertical
-import platform.UIKit.UICollectionViewScrollPositionTop
 import platform.UIKit.UIColor
 import platform.UIKit.UIControlEventValueChanged
 import platform.UIKit.UIEdgeInsetsMake
 import platform.UIKit.UIRefreshControl
 import platform.UIKit.UIScrollView
+import platform.UIKit.UITableView
+import platform.UIKit.UITableViewAutomaticDimension
+import platform.UIKit.UITableViewCell
+import platform.UIKit.UITableViewCellSeparatorStyle.UITableViewCellSeparatorStyleNone
+import platform.UIKit.UITableViewCellStyle
+import platform.UIKit.UITableViewDataSourceProtocol
+import platform.UIKit.UITableViewDelegateProtocol
+import platform.UIKit.UITableViewRowAnimationNone
+import platform.UIKit.UITableViewScrollPosition
 import platform.UIKit.UIView
 import platform.UIKit.indexPathForItem
 import platform.UIKit.item
 import platform.darwin.NSInteger
 import platform.darwin.NSObject
 
 internal open class UIViewLazyList(
-  private var collectionViewFlowLayout: UICollectionViewFlowLayout = UICollectionViewFlowLayout().apply {
-    estimatedItemSize = UICollectionViewFlowLayoutAutomaticSize.readValue()
-  },
-  internal val collectionView: UICollectionView = UICollectionView(
+  internal val tableView: UITableView = UITableView(
     CGRectZero.readValue(),
-    collectionViewFlowLayout,
   ),
-) : WindowedLazyList<UIView>(UICollectionViewListUpdateCallback(collectionView)), ChangeListener {
+) : LazyList<UIView>, ChangeListener {
```

Comment:
The goal of `WindowedLazyList` was to have `UIViewLazyList` and `ViewLazyList` to share the bulk of their implementations. They are fairly similar (e.g., view based, platform provides a recycling list view that allows atomic updates). The integration of `WindowedLazyList` with `UIViewLazyList` wasn't complete, namely that `UICollectionViewListUpdateCallback` didn't yet delegate to atomic updates.

What are your thoughts on continuing down that path vs switching approach as you did here?

Reply:
There's this issue from last week that has more info: https://github.com/cashapp/redwood/issues/1498

Reply:
This new class started out as a copy-paste of WindowedLazyList, addressing problems on iOS by adding precise updates.

Are precise updates useful for RecyclerView? If precise updates will make RecyclerView faster, then we should follow-up by migrating RecyclerView to LazyListUpdateProcessor (and delete the old WindowedLazyList). If precise updates are not necessary for RecyclerView, then we should not do precise updates; it’s a bit of work to provide them.

Reply:
Reading that issue makes it clear to me that this new thing will benefit both platforms. I’d still like to land it one-platform-at-a-time, mostly because our test coverage here is lacking.

Reply:
That sounds good. Do you mind adding a `TODO` here (or manually create an issue) so that we remember that these two mechanisms should ideally be merged?

Reply:
Here’s a tracking issue: https://github.com/cashapp/redwood/issues/1552